muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/actions/main-function/records/summary.ts

```diff
@@ -36,9 +36,30 @@ let updateSummary: UpdateSummaryTriggerType = function updateSummary() {
       let assignmentsDone: any = [];
       let coursesDone: any = [];
 
-      /* Students study time */
+      /* Student's study time */
       let studentsDetails: any = await promisify(mApi().user.students.read(pyramusId), 'callback')();
 
+      /* Student's user groups */
+      let studentsUserGroups: any = await promisify(mApi().usergroup.groups.read({ userIdentifier: pyramusId }), 'callback')();
+
+      let studentsStudentCouncelors: any = [];
+
+      /*
+        We need to filter student's usergroups that are guidance groups, then we fetch guidance councelors
+        of those usergroups and push the result to studentsGuidanceCouncelors array
+      */
+      if (studentsUserGroups && studentsUserGroups.length) {
+        studentsUserGroups.filter((studentsUserGroup: any) => studentsUserGroup.guidanceGroup == true).forEach(function (studentsUserGroup: any) {
+          mApi().usergroup.groups.staffMembers.read(studentsUserGroup.id, {properties: 'profile-phone,profile-vacation-start,profile-vacation-end'}).callback(function (err: any, result: any) {
+            result.forEach(function (studentsStudentCouncelor: any) {
```

Comment:
The result set could probably be sorted by last name so the names wouldn't be in random order in the UI.